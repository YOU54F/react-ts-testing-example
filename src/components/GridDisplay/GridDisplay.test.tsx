import { string } from 'prop-types';
import * as React from 'react';
import renderer from 'react-test-renderer';
import { GridDisplay } from './GridDisplay';

describe('Grid Display', () => {
  
  it('can render a grid with a no strings in the array', () => {
    const tree = renderer.create(<GridDisplay gridItems={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('can render a grid with a single string in the array', () => {
    const tree = renderer.create(<GridDisplay gridItems={['foo']} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('can render a grid with a 2 strings in the array', () => {
    const tree = renderer.create(<GridDisplay gridItems={["foo","bar"]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can render a grid with a collection or urls in the array', () => {
    const tree = renderer.create(<GridDisplay
          gridItems={[
            "https://loremflickr.com/100/150",
            "An",
            "https://loremflickr.com/100/150",
            " ",
            "https://loremflickr.com/100/150",
            " ",
            "https://loremflickr.com/100/150",
            "Example",
            "https://loremflickr.com/100/150",
            " ",
            "https://loremflickr.com/100/150",
            " ",
            "https://loremflickr.com/100/150",
            "Website",
            "https://loremflickr.com/100/150"
          ]}
        />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
