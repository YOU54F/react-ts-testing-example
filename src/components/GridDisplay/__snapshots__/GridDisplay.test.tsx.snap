// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Display can render a grid with a 2 strings in the array 1`] = `
<div
  className="grid-display-container"
>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "0s",
      }
    }
  >
    <p
      className="grid-text"
    >
      foo
    </p>
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "0.5s",
      }
    }
  >
    <p
      className="grid-text"
    >
      bar
    </p>
  </div>
</div>
`;

exports[`Grid Display can render a grid with a collection or urls in the array 1`] = `
<div
  className="grid-display-container"
>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "0s",
      }
    }
  >
    <img
      alt="grid"
      className="grid-image"
      id="grid-item-0"
      src="https://loremflickr.com/100/150"
    />
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "0.5s",
      }
    }
  >
    <p
      className="grid-text"
    >
      An
    </p>
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "1s",
      }
    }
  >
    <img
      alt="grid"
      className="grid-image"
      id="grid-item-2"
      src="https://loremflickr.com/100/150"
    />
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "1.5s",
      }
    }
  >
    <p
      className="grid-text"
    >
       
    </p>
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "2s",
      }
    }
  >
    <img
      alt="grid"
      className="grid-image"
      id="grid-item-4"
      src="https://loremflickr.com/100/150"
    />
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "2.5s",
      }
    }
  >
    <p
      className="grid-text"
    >
       
    </p>
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "3s",
      }
    }
  >
    <img
      alt="grid"
      className="grid-image"
      id="grid-item-6"
      src="https://loremflickr.com/100/150"
    />
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "3.5s",
      }
    }
  >
    <p
      className="grid-text"
    >
      Example
    </p>
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "4s",
      }
    }
  >
    <img
      alt="grid"
      className="grid-image"
      id="grid-item-8"
      src="https://loremflickr.com/100/150"
    />
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "4.5s",
      }
    }
  >
    <p
      className="grid-text"
    >
       
    </p>
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "5s",
      }
    }
  >
    <img
      alt="grid"
      className="grid-image"
      id="grid-item-10"
      src="https://loremflickr.com/100/150"
    />
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "5.5s",
      }
    }
  >
    <p
      className="grid-text"
    >
       
    </p>
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "6s",
      }
    }
  >
    <img
      alt="grid"
      className="grid-image"
      id="grid-item-12"
      src="https://loremflickr.com/100/150"
    />
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "6.5s",
      }
    }
  >
    <p
      className="grid-text"
    >
      Website
    </p>
  </div>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "7s",
      }
    }
  >
    <img
      alt="grid"
      className="grid-image"
      id="grid-item-14"
      src="https://loremflickr.com/100/150"
    />
  </div>
</div>
`;

exports[`Grid Display can render a grid with a no strings in the array 1`] = `
<div
  className="grid-display-container"
/>
`;

exports[`Grid Display can render a grid with a single string in the array 1`] = `
<div
  className="grid-display-container"
>
  <div
    className="grid-item fade-out fade-in"
    style={
      Object {
        "animationDelay": "0s",
      }
    }
  >
    <p
      className="grid-text"
    >
      foo
    </p>
  </div>
</div>
`;
